Dockerfile

```diff
@@ -110,29 +80,83 @@ ENV GO111MODULE on
 RUN curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
   && echo "$GOLANG_DOWNLOAD_SHA256 golang.tar.gz" | sha256sum -c - \
   && tar -C /usr/local -xzf golang.tar.gz \
-  && ln -sf /usr/local/go/bin/go /usr/local/bin \
-  && rm golang.tar.gz
+  && ln -sf /usr/local/go/bin/go /usr/local/bin
 
 RUN curl -fsSL "$GOSEC_DOWNLOAD_URL" -o gosec.tar.gz \
   && echo "$GOSEC_DOWNLOAD_SHA256 gosec.tar.gz" | sha256sum -c - \
-  && mkdir gosec && tar -C gosec -zxf gosec.tar.gz \
-  && mv gosec/gosec /usr/local/bin \
-  && rm -rf gosec gosec.tar.gz
+  && mkdir gosec && tar -C gosec -zxf gosec.tar.gz
 
 
 ### sift
 ENV SIFT_VERSION v0.9.0
 
-RUN go get github.com/svent/sift@${SIFT_VERSION} \
-  && mv /root/go/bin/sift /usr/local/bin \
-  && rm -rf go ~/.cache
+RUN go get github.com/svent/sift@${SIFT_VERSION}
 
 
 ### semgrep
 # https://semgrep.dev
 ENV SEMGREP_VERSION 0.36.0
 
-RUN pip3 install semgrep==${SEMGREP_VERSION}
+RUN pip3 install --user --no-cache-dir semgrep==${SEMGREP_VERSION}
+
+
+### Ruby
+COPY Gemfile Gemfile.lock ./
+RUN bundle install --deployment --without development:test
+
+
+
+FROM ruby:2.7.2-slim
```

Comment:
Should we make 2.7.2 a docker environment variable here so that the top level image and this slim image are the same ruby versions?

Reply:
Also, can you add in the SHA hash here like we have at the top?

Reply:
Yup - should I update the hash to get the newest version of the image?

Reply:
I think leave the one up top as is for now. We can go back and update when we do our regular patching and testing. Latest docker SHA commit for the slim image should be fine.

Reply:
Ah, it seems that you can't parameterize the `FROM` directive like that.

Reply:
Thanks for trying :(